tsdb/engine/tsm1/reader.go

```diff
@@ -263,6 +273,15 @@ func NewTSMReader(f *os.File, options ...tsmReaderOption) (*TSMReader, error) {
 		return nil, err
 	}
 
+	if nil != t.parseFileNameFunc {
```

Comment:
Alternatively, we could fallback to using the default parsing function if `WithParseFileNameFunc` is not used.

Reply:
That would break tests.

Reply:
That breaks tests.  The whole parsing function pointer could now be a boolean flag (parse/no_parse) and the function hard-coded, if that seems cleaner

Reply:
No_parse for some tests, parse by default